api/src/domain/tax-filings/taxFilingsInterfaceFactory.test.ts

```diff
@@ -140,83 +140,81 @@ describe("TaxFilingsInterfaceFactory", () => {
         });
       });
 
-      it("returns lookup response when lookup returns API_ERROR", async () => {
+      it("returns lookup response when onboarding returns SUCCESS", async () => {
+        const filingData = generateTaxFiling({});
+        taxFilingClient.onboarding.mockResolvedValue({
+          state: "SUCCESS",
+          filings: [filingData],
+        } as TaxFilingOnboardingResponse);
         taxFilingClient.lookup.mockResolvedValue({
-          state: "API_ERROR",
-          filings: [],
+          state: "SUCCESS",
+          filings: [filingData],
         } as TaxFilingLookupResponse);
         expect(await taxFilingInterface.onboarding({ userData, ...taxIdBusinessName })).toEqual({
           ...userData,
           taxFilingData: {
             ...userData.taxFilingData,
-            state: "API_ERROR",
-            registeredISO: undefined,
+            state: "SUCCESS",
             businessName: taxIdBusinessName.businessName,
+            filings: [filingData],
             lastUpdatedISO: currentDate.toISOString(),
           },
         });
-        expect(taxFilingClient.onboarding).not.toHaveBeenCalled();
       });
+    });
 
-      it("returns lookup response when lookup returns FAILED", async () => {
-        taxFilingClient.lookup.mockResolvedValue({
-          state: "FAILED",
-          filings: [],
-        } as TaxFilingLookupResponse);
-        expect(await taxFilingInterface.onboarding({ userData, ...taxIdBusinessName })).toEqual({
-          ...userData,
-          taxFilingData: {
-            ...userData.taxFilingData,
-            state: "FAILED",
+    describe("returns onboarding response", () => {
+      it("passes through the API_ERROR response", async () => {
+        const filingData = generateTaxFiling({});
+        const userData = generateUserData({
+          taxFilingData: generateTaxFilingData({
+            state: undefined,
+            lastUpdatedISO: undefined,
             registeredISO: undefined,
-            businessName: taxIdBusinessName.businessName,
-            lastUpdatedISO: currentDate.toISOString(),
-          },
+            filings: [filingData],
```

Comment:
why are we starting with filingData for the user before the onboarding/lookup happens? Would this state ever actually exist?

Reply:
same comment on all these